File: jmlparser-jml-tools/src/main/java/com/github/jmlparser/lint/rules/LintProblemMeta.java
package com.github.jmlparser.lint.rules;

import com.github.javaparser.ast.Node;
import com.github.javaparser.ast.nodeTypes.NodeWithTokenRange;
import com.github.jmlparser.lint.LintProblem;

/**
 * @author Alexander Weigl
 * @version 1 (21.10.22)
 */
public record LintProblemMeta(String id, String message, String level) {

    public LintProblem create(NodeWithTokenRange<Node> n) {
        return new LintProblem(level(), message(), n.getTokenRange().orElse(null), null, id, ruleId);
    }
}

File: tools/lint/src/main/kotlin/io/github/jmltoolkit/lint/rules/LintProblemMeta.kt
package io.github.jmltoolkit.lint.rules

import com.github.javaparser.ast.Node
import com.github.javaparser.ast.nodeTypes.NodeWithTokenRange
import io.github.jmltoolkit.lint.LintProblem

/**
 * @author Alexander Weigl
 * @version 1 (21.10.22)
 */
@JvmRecord
data class LintProblemMeta(val id: String, val message: String, val level: String) {
    fun create(n: NodeWithTokenRange<Node?>): LintProblem {
        return LintProblem(level, message, n.tokenRange.orElse(null), null, id, "")
    }
}
